rmgpy/reaction.py

```diff
@@ -1144,6 +1144,99 @@ def ensure_species(self, reactant_resonance=False, product_resonance=True):
         except AttributeError:
             pass
 
+    def check_collision_limit_violation(self, t_min, t_max, p_min, p_max):
+        """
+        Warn if a core reaction violates the collision limit rate in either the forward or reverse direction
+        at the relevant extreme T/P conditions. Assuming a monotonic behaviour of the kinetics.
+        Returns a list with the reaction object and the direction in which the violation was detected.
+        """
+        conditions = [[t_min, p_min]]
+        if t_min != t_max:
+            conditions.append([t_max, p_min])
+        if self.kinetics.isPressureDependent() and p_max != p_min:
+            conditions.append([t_min, p_max])
+            if t_min != t_max:
+                conditions.append([t_max, p_max])
+        logging.debug("Checking whether reaction {0} violates the collision rate limit...".format(self))
+        violator = None
+        kf_list = []
+        kr_list = []
```

Comment:
With the newest simplification, I don't thing kr_list is being used (are you missing the code that handles checking the reverse reaction? I like the direction of this code simplification, but I think it is only checking the forward direction now.

Reply:
Can you take a look at line 1180 in this file? I think `kr_list` is used as expected

https://github.com/ReactionMechanismGenerator/RMG-Py/pull/1361/files/fe94464ff3ad69d8b4c12e9950cc39e1dafede75#diff-eb892c0ffd531a3aa16ba76563c6ccc6R1180

Reply:
Does anything append on to `kr_list` before line 1180 though? Maybe I am missing something

Reply:
You're right. I'll fix it.